azurelinuxagent/common/cgroup.py

```diff
@@ -13,104 +13,139 @@
 # limitations under the License.
 #
 # Requires Python 2.6+ and Openssl 1.0+
-
+import os
 import re
 
+from azurelinuxagent.common import logger
 from azurelinuxagent.common.exception import CGroupsException
+from azurelinuxagent.common.future import ustr
 from azurelinuxagent.common.osutil import get_osutil
+from azurelinuxagent.common.utils import fileutil
 
-re_user_system_times = re.compile('user (\d+)\nsystem (\d+)\n')
+re_user_system_times = re.compile(r'user (\d+)\nsystem (\d+)\n')
 
 # The metric classes (Cpu, Memory, etc) can all assume that CGroups is enabled, as the CGroupTelemetry
 # class is very careful not to call them if CGroups isn't enabled. Any tests should be disabled if the osutil
 # is_cgroups_support() method returns false.
 
 
 class CGroup(object):
-    def __init__(self):
-        pass
-
-    def _get_cgroup_file(self, hierarchy, file_name):
-        return os.path.join(self.cgroups[hierarchy], file_name)
+    def __init__(self, extension_name, cgroup_path, controller_type):
+        self.name = extension_name
+        self.path = cgroup_path
+        self.controller = controller_type
 
+    def _get_cgroup_file(self, file_name):
+        return os.path.join(self.path, file_name)
 
-    def get_file_contents(self, hierarchy, file_name):
+    def get_file_contents(self, controller, file_name):
         """
-        Retrieve the value of a parameter from a hierarchy.
+        Retrieve the value of a parameter from a controller.
```

Comment:
>  Retrieve the value of a parameter from a controller

that is what get_parameter does... this comment should probably state that the function returns the contents of a file